### ISSUE TITLE
[51] Are Large-scale Datasets Necessary for Self-Supervised Pre-training? (SplitMask)

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#79]
[paper](https://arxiv.org/pdf/2112.10740.pdf)

target dataset 에 맞춰서 pretraining 하는 게 더 좋은 것 같은데?
굳이 large-scale pretraining 필요 있어?
라고 주장하는 논문

실제로 MaskFeat(#73) 을 보면 Kinetic-400 에서도 비슷한 현상이 있다. ImageNet-21k pretrain 도 힘 못씀.

BEiT 방식이나 제안하는 Denoising AE 방식이 pretraining 에 효과적이라고 주장.
![image](https://user-images.githubusercontent.com/16400591/147720515-cd755854-c733-45dd-a458-9f8f79640aba.png)

# SplitMask
![image](https://user-images.githubusercontent.com/16400591/147720674-b1bd6fc6-7e9c-4e6b-a5c1-f1b885dd9995.png)

간단함. 
1. decoder는 2layer.
2. mask 는 절반 randon 생성
3. 생성한 mask 로 not mask 생성
4. encoder 는 전부 통과
5. decoder 2개에서 각각 recon loss
6. Contrastive loss (Info NCE)

contrastive loss 는 x_a, x_b 를 각각 얻었다고 가정했을 때, 다음과 같이 정의됨
![image](https://user-images.githubusercontent.com/16400591/147721884-e512edea-10db-452f-b794-05a866887e50.png)

N은 negative. 같은 batch 내에서 다른 sample 들.
Temperature 는 hyperparameter (== 0.2)

Stanford-Cars, ClipArt, Sketch and Paintings 와 같이 작은 set 들에 대해서는 75% masking 을 하는게 오버피팅을 막았다고 함.

# Analysis
## Sample Efficiency & Learning using non object-centric images
ImageNet pretrain 후 iNaturalist-2019 에 transfer learning 수행.
![image](https://user-images.githubusercontent.com/16400591/147720614-a425eb5e-b939-4eb8-b178-9dcf393164b2.png)

데이터를 줄여서 써보면서 epoch 수도 낮췄음 
적은 데이터에서는 pretraining 도 overfit 하기 때문
![image](https://user-images.githubusercontent.com/16400591/147722648-9ea36954-b35a-4742-a2d0-ef41e99a4f0b.png)

iNaturalist-2019 set 기준 사실 5% 까지만 써도 충분하더라.
![image](https://user-images.githubusercontent.com/16400591/147722687-7818f030-50cc-4407-9874-00019c60dc0c.png)

COCO 는 Imagenet 처럼 object-centric image 가 아님.
COCO 수치는 DINO 나 BEiT 에서는 잘 못하더라. ~(#66 )을 같이 쓴다면?!~

BEiT 에 tokenizer 변경해 보면서 small set 성능 관찰
![image](https://user-images.githubusercontent.com/16400591/147722880-d91306bf-32ce-44e4-8360-5bea342c89cb.png)
굳이 DALL-E tokenizer 안 써도 된다고 주장.

# Experiments
사용한 데이터와 pretraining epoch들은 다음과 같음
![image](https://user-images.githubusercontent.com/16400591/147722931-9b3a44cd-3ef6-4c44-8475-834e869c3396.png)

## COCO Detection & Instance Segmentation
![image](https://user-images.githubusercontent.com/16400591/147722979-19b3b58d-a899-4cfc-bcfd-1e638109ff15.png)

## ADE-20K
![image](https://user-images.githubusercontent.com/16400591/147722996-17312cbe-f415-48a2-81db-d0385d8da7bb.png)

## 기타 small sets
![image](https://user-images.githubusercontent.com/16400591/147723054-65eb70f8-fbbe-4a9d-949d-bd73fa7cefac.png)

# 필자 의견
단순 pretraining 뿐 아니라, task-wise data pretraining 이 함께 도는 모델을 연구할 필요성이 느껴짐.

# Appendix
## Appendix A SplitMask vs BEiT
![image](https://user-images.githubusercontent.com/16400591/147722277-f811806e-fa99-4e28-b198-b2107fb80700.png)

## Appendix B  Encoder-Decoder vs BEiT
layer-wise feature 들에 linear probing 수행해 봄
![image](https://user-images.githubusercontent.com/16400591/147722362-a1a7c0d7-1cda-4368-9c9b-b54013f86847.png)

## Appendix C Overfitting during pre-training
pre-training scheme 를 맞춰본다면,
Imagenet에서 300 epoch pretraining == stanford cars 에서 45k epoch
이 맞음
근데, overfitting 이 발생해서 5k epoch 으로 만족
![image](https://user-images.githubusercontent.com/16400591/147722435-676feba8-c00b-431c-a493-12d9bffd2e45.png)

## Appendix D Hyperparams
![image](https://user-images.githubusercontent.com/16400591/147722226-23423ef3-02b9-4ac2-aa0b-a8ecb2e81ef6.png)

**Comment by wengdunfang** [comment]
Have you implemented the code?

**Comment by dhkim0225** [comment]
@wengdunfang Not Yet.
